linera-core/src/updater.rs

```diff
@@ -200,47 +202,24 @@ where
     A: ValidatorNode + Clone + 'static,
     S: Storage + Clone + Send + Sync + 'static,
 {
-    async fn send_optimized_certificate(
-        &mut self,
-        certificate: &Certificate,
-        delivery: CrossChainMessageDelivery,
-    ) -> Result<Box<ChainInfo>, NodeError> {
-        if certificate.is_signed_by(&self.remote_node.name) {
-            let result = self
-                .remote_node
-                .handle_lite_certificate(certificate.lite_certificate(), delivery)
-                .await;
-            match result {
-                Err(NodeError::MissingCertificateValue) => {
-                    warn!(
-                        "Validator {} forgot a certificate value that they signed before",
-                        self.remote_node.name
-                    );
-                }
-                _ => {
-                    return result;
-                }
-            }
-        }
-        self.remote_node
-            .handle_certificate(certificate.clone(), vec![], delivery)
-            .await
-    }
-
     async fn send_certificate(
         &mut self,
         certificate: Certificate,
         delivery: CrossChainMessageDelivery,
     ) -> Result<Box<ChainInfo>, NodeError> {
         let result = self
-            .send_optimized_certificate(&certificate, delivery)
+            .remote_node
+            .handle_optimized_certificate(&certificate, delivery)
             .await;
 
         match &result {
             Err(original_err @ NodeError::BlobsNotFound(blob_ids)) => {
+                self.remote_node
+                    .check_blobs_not_found(&certificate, blob_ids)?;
+
                 let blobs = self
                     .local_node
-                    .find_missing_blobs(&certificate, blob_ids, certificate.inner().chain_id())
+                    .find_missing_blobs(blob_ids.clone(), certificate.inner().chain_id())
```

Comment:
If I didn't know better I'd say that we should check with a local node to handle certificates and download any missing blobs from a remote node. Maybe the `handle_optimized_certificate` is a bad name? For some reason, If I see a method called `handle_*` I imagine that it's an internal logic where I ask to carry out a job for me but here it looks like a local node asks a remote to handle a cert it (local) just created?

Reply:
Maybe, but then we should change it everywhere. Currently `RemoteNode` (and the gRPC node client implementation and everything) uses the term `handle_` for all of these.